framework/src/main/java/org/checkerframework/common/accumulation/AccumulationAnnotatedTypeFactory.java

```diff
@@ -248,6 +320,21 @@ public QualifierHierarchy createQualifierHierarchy(MultiGraphFactory factory) {
      *        |
      *      bottom
      * </pre>
+     *
+     * Predicate subtyping is defined as follows:
+     *
+     * <ul>
+     *   <li>An accumulator is a subtype of a predicate if substitution from the accumulator to the
+     *       predicate makes the predicate true. For example, {@code Acc(A)} is a subtype of {@code
+     *       AccPred("A || B")}, because when A is replaced with {@code true} and B is replaced with
+     *       {@code false}, the resulting boolean formula evaluates to true.
+     *   <li>A predicate P is a subtype of an accumulator iff after converting the accumulator into
+     *       a predicate representing the conjunction of its elements, P is a subtype of that
```

Comment:
Does this mean that after converting the accumulator to a predicate, `P` must _equal_ that predicate (i.e., it falls into the next case of predicate subtyping)?

Reply:
In the default implementation, yes, *but* if the accumulation checker writer has implemented a more complex predicate subtyping rule that would also be taken into account.